pilot/pkg/serviceregistry/external/controller_test.go

```diff
@@ -97,19 +83,19 @@ func TestController(t *testing.T) {
 			Resolution: networking.ServiceEntry_STATIC,
 		},
 	}
-	expectedCount := 7 // 1 service + 6 instances
+	expectedCount := int64(7) // 1 service + 6 instances
 
 	_, err = configController.Create(cfg)
 	if err != nil {
 		t.Fatalf("error occurred crearting ServiceEntry config: %v", err)
 	}
 
 	if err := retry.UntilSuccess(func() error {
-		if c := getCountAndReset(); c != expectedCount {
-			return fmt.Errorf("got %d notifications from controller, want %d", c, 7)
+		if gotcount := atomic.AddInt64(&count, 0); gotcount != expectedCount {
+			return fmt.Errorf("got %d notifications from controller, want %d", gotcount, expectedCount)
 		}
 		return nil
-	}, retry.Delay(1*time.Second), retry.Timeout(20*time.Second)); err != nil {
+	}, retry.Delay(50*time.Millisecond), retry.Timeout(5*time.Second)); err != nil {
```

Comment:
Let's make the retry timeout a bit higher ... we have a lot of issues with sluggishness on prow.  Maybe 10?